# 01 숫자형

### 정수형

In [2]:
a=123
a=-178
a=0

### 실수형

In [3]:
a=1.2
a=-3.45

### 8진수

In [4]:
a=0o177
print(a)

127


### 16진수

In [5]:
a=0x8ff
b=0xabc
print(b)

2748


### 사칙연산

In [1]:
a=3
b=4
a+b

7

In [2]:
a-b

-1

In [3]:
a*b

12

In [4]:
a/b

0.75

### x의 y제곱을 나타내는 ** 연산자 (x=a, y=b)

In [10]:
a=3
b=4
a**b

81

### 나눗셈 후 나머지를 리턴하는 % 연산자
<img src="https://wikidocs.net/images/page/12/02_1_mod1.png" align="left"></img>

In [11]:
3%7

3

In [8]:
7%3

1

### 나눗셈 후 몫을 리턴하는 // 연산자
<img src="https://wikidocs.net/images/page/12/02_1_mod2.png" align="left"></img>

In [23]:
7/4

1.75

In [24]:
7//4

1

1.75에서 몫에 해당하는 정숫값 1만 리턴하는 것을 확인할 수 있다.

----

# 02 문자열 자료형

In [11]:
"hello world"

'hello world'

"a" >> 'a'

In [12]:
'hello world'

'hello world'

'a >> 'a'

### 문자열 안에 작은따옴표나 큰따옴표를 포함시키고 싶을 때

In [13]:
"Python's favorite food is perl"

"Python's favorite food is perl"

"a's" >> "a's"

In [18]:
'"Python is very easy." he says.'

'"Python is very easy." he says.'

' "abc" ' >> ' "abc" '

In [19]:
'Python\'s favorite food is perl'

"Python's favorite food is perl"

'a\'s' >> "a's"

In [2]:
"\"Python is very easy.\" he says."

'"Python is very easy." he says.'

"\"abc.\"defg" >> '"abc."defg'

### 줄바꿈

In [8]:
"Life is too short\nYou need python"

'Life is too short\nYou need python'

In [5]:
#1. 줄을 바꾸기 위한 이스케이프 코드 \n 삽입하기
multiline = "Life is too short\nYou need python"

In [6]:
print(multiline)

Life is too short
You need python


In [7]:
#2. 연속된 작은따옴표 3개 또는 큰따옴표 3개 사용하기
multiline = '''
Life is too short
You need python
'''

In [8]:
multiline

'\nLife is too short\nYou need python\n'

In [9]:
print(multiline)


Life is too short
You need python



### 문자열 연산하기

In [12]:
#문자열 더해서 연결하기

head = "Python"
tail = " is fun!"
head + tail

'Python is fun!'

In [13]:
#문자열 곱하기

a='python'
a*5

'pythonpythonpythonpythonpython'

### 문자열 길이 구하기

In [16]:
a="life is too short"
len(a)

17

### 문자열 인덱싱 (파이썬은 0부터 숫자를 센다)

In [17]:
a = "Life is too short, You need Python"
a[3]

'e'

In [18]:
a[12]

's'

### - 는 뒤에서 부터

In [19]:
a[-1]

'n'

In [20]:
a[-7]

' '

### 문자열 슬라이싱 (end 인덱스 +1 해야한다)

In [25]:
a = "Life is too short, You need Python"
a[0:3]

'Lif'

In [29]:
a[0:4]

'Life'

In [30]:
a[8:17]

'too short'

In [32]:
a[0:4]==a[0:5]

False

In [33]:
a[19:]

'You need Python'

In [34]:
a[:17]

'Life is too short'

In [36]:
a[:]

'Life is too short, You need Python'

In [38]:
a[19:-7]

'You need'

### 슬라이싱으로 문자열 나누기

In [40]:
a = "20230331Rainy"
date = a[:8]
weather = a[8:]
date, weather

('20230331', 'Rainy')

In [52]:
a = "20230331Rainy"
year = a[:4]
day = a[4:8]
weather = a[8:]
year

'2023'

In [53]:
day

'0331'

In [54]:
weather

'Rainy'

In [51]:
#문자열 수정

a='pithon'
a[1]=y
a

#당연히 오류가 발생한다. 문자열의 요솟값은 바꿀 수 있는 값이 아니기 때문이다
#(그래서 문자열을 ‘변경 불가능한(immutable) 자료형’이라고도 부른다).

NameError: name 'y' is not defined

In [44]:
a[:1]

'p'

In [45]:
a[2:]

'thon'

In [47]:
b= a[:1]+'y'+a[2:]
b

'python'

### 문자열 포매팅

In [19]:
#숫자바로대입 - %d

"I eat %d apples."%3

'I eat 3 apples.'

In [20]:
#문자열 바로대입 - %s

"I eat %s apples."%"five"

'I eat five apples.'

In [59]:
"I eat %d apples."%삼

NameError: name '삼' is not defined

In [18]:
#숫자값을 나타내는 변수로 대입

number=3
"I eat %d apples."%number

'I eat 3 apples.'

In [17]:
#2개 이상의 값 넣기

number = 10
day ='three'
"I ate %d apples. so I was sick for %s days." % (number, day)

'I ate 10 apples. so I was sick for three days.'

In [16]:
#d>>숫자 안됌, s>>문자로 인지가 됌

"I have %s apples" %3

'I have 3 apples'

In [70]:
"오늘의 습도는%d%% 입니다." %70

'오늘의 습도는70% 입니다.'

### 포맷 코드와 숫자 함께 사용하기

In [15]:
#1.정렬과 공백

"%10s" % "hi"

'        hi'

In [72]:
"%-10sjane." % 'hi'

'hi        jane.'

In [21]:
#2. 소수점 표현하기

"%0.4f" % 3.42134234

'3.4213'

In [74]:
"%10.4f" % 3.42134234

'    3.4213'

### format 함수를 사용한 포매팅

In [77]:
#숫자 바로 대입하기

"I eat {} apples".format(3)

'I eat 3 apples'

In [76]:
# 문자열 바로 대입하기

"I eat {} apples".format('five')

'I eat five apples'

In [78]:
#숫자 값을 가진 변수로 대입하기

number=10
"I eat {} apples".format(number)

'I eat 10 apples'

In [81]:
#2개 이상의 값 넣기

number = 10
day = "three"
"I ate {} apples. so I was sick for {} days.".format(number, day)

'I ate 10 apples. so I was sick for three days.'

In [22]:
#이름 {number} 형태로 넣기

"I ate {number} apples. so I was sick for {day} days.".format(number=10, day=3)

'I ate 10 apples. so I was sick for 3 days.'

In [23]:
#인덱스와 이름을 혼용해서 넣기 
    
"I ate {0} apples. so I was sick for {day} days.".format(10, day=3)

'I ate 10 apples. so I was sick for 3 days.'

In [24]:
#소수점 표현하기

y = 3.42134234
"{0:0.4f}".format(y)

'3.4213'

### f 문자열 포매팅 (강추) - 문자열 앞에 f 접두사를 붙이면 f 문자열 포매팅 기능을 사용할 수 있다.

In [85]:
name = '홍길동'
age = 30
f'나의 이름은 {name}입니다. 나이는 {age}입니다.'

'나의 이름은 홍길동입니다. 나이는 30입니다.'

In [25]:
##f 문자열 포매팅은 표현식을 지원하기 때문에 다음과 같은 것도 가능하다.

age = 30
f'나는 내년이면 {age + 1}살이 된다.'

'나는 내년이면 31살이 된다.'

In [87]:
##딕셔너리는 f 문자열 포매팅에서 다음과 같이 사용할 수 있다.

d = {'name':'홍길동', 'age':30}
f'나의 이름은 {d["name"]}입니다. 나이는 {d["age"]}입니다.'

'나의 이름은 홍길동입니다. 나이는 30입니다.'

### 문자열 관련 함수들

In [88]:
#문자 개수 세기 - count

a = "hobby"
a.count('b')

2

In [89]:
#위치 알려 주기 1 - find

a = "Python is the best choice"
a.find('b')

14

In [90]:
#위치 알려 주기 2 - index 인덱스는 오류를 보여준다

a = "Life is too short"
a.index('t')

8

In [91]:
a.index('k')

ValueError: substring not found

In [92]:
#문자열 삽입 - join

",".join('abcd')

'a,b,c,d'

In [93]:
#소문자를 대문자로 바꾸기 - upper

a = "hi"
a.upper()

'HI'

In [95]:
#대문자를 소문자로 바꾸기 - lower

a="HI"
a.lower()

'hi'

In [96]:
#왼쪽 공백 지우기 - lstrip

a= " hi "
a.lstrip()

'hi '

In [97]:
#오른쪽 공백 지우기 - rstrip

a= " hi "
a.rstrip()

' hi'

In [98]:
#양쪽 공백 지우기 - strip

a= " hi "
a.strip()

'hi'

In [27]:
#문자열 바꾸기 - replace
#replace(바뀔_문자열, 바꿀_문자열)처럼
#사용해서 문자열 안의 특정한 값을 다른 값으로 치환

a = "Life is too short"
a.replace('Life','Your leg')

'Your leg is too short'

In [100]:
a

'Life is too short'

In [28]:
#문자열 나누기 - split
#split 함수는 a.split()처럼 괄호 안에
#아무 값도 넣어 주지 않으면 공백([Space]], [Tab], [Enter])을
#기준으로 문자열을 나누어 준다.

a = "Life is too short"
a.split()

['Life', 'is', 'too', 'short']

In [102]:
#괄호 안에 특정 값이 있을 경우에는 괄호 안의 값을
#구분자로 해서 문자열을 나누어 준다.
#이렇게 나눈 값은 리스트에 하나씩 들어간다. 

b='a:b:c:d'
b.split(':')

['a', 'b', 'c', 'd']

# 자료형의 값을 저장하는 공간, 변수
##### 파이썬에서 사용하는 변수는 객체를 가리키는 것이라고도 말할 수 있다. 객체란 우리가 지금까지 보아 온 자료형의 데이터(값)와 같은 것을 의미하는 말이다

In [86]:
a=1
a, type(a)

(1, int)

In [87]:
#문자데이터

b='python'
b, type(b)

('python', str)

In [88]:
#리스트데이터

c=[1,2,3]
c, type(c)

([1, 2, 3], list)

In [89]:
#id는 변수가 가리키고 있는 객체의 주소 값을 리턴하는 파이썬의 내장 함수

id(a)

140715559457576

# 리스트를 복사하고자 할 때

In [90]:
#문자열을 복사
a=[1,2,3]
b=a

In [91]:
id(a), id(b)

(2466400779904, 2466400779904)

In [92]:
a is b ## a와 b가 가리키는 객체가 같을까?

True

In [93]:
a

[1, 2, 3]

In [94]:
a[1]=4

In [95]:
a

[1, 4, 3]

In [96]:
b

[1, 4, 3]

## 두 변수의 값은 같지만, 가리키는 객체는 서로 다르다는 것을 알 수 있다.

In [43]:
num1=10
num1

10

In [44]:
num2=num1
num1, num2

(10, 10)

In [45]:
num1=11
num1, num2

(11, 10)

In [46]:
#[:] 이용하기

a = [1, 2, 3]
b = a[:]
a[1] = 4
a

[1, 4, 3]

In [47]:
b

[1, 2, 3]

In [97]:
#copy 모듈 이용하기

from copy import copy
a=[1,2,3]
a

[1, 2, 3]

In [98]:
b = copy(a)
id(a), id(b)

(2466400643072, 2466400771904)

In [99]:
b is a

False

### 변수를 만드는 여러가지 방법

In [100]:
#튜플로 a, b에 값을 대입할 수 있다.

a, b = ('python', 'life')

In [106]:
a,b

('python', 'life')

In [107]:
#튜플은 괄호를 생략해도 된다.

(a, b) = 'python', 'life'

In [108]:
a,b

('python', 'life')

In [109]:
#리스트로 변수를 만들 수도 있다.

[a, b] = ['python', 'life']

In [57]:
a,b

('python', 'life')

In [110]:
#여러 개의 변수에 같은 값을 대입할 수도 있다. 

a = b = 'python'

In [111]:
a,b

('python', 'python')

# 리스트 자료형

In [1]:
odd = [1,3,5,7,9]
odd

[1, 3, 5, 7, 9]

In [2]:
type(odd)

list

In [6]:
##리스트안에는 여러 객체들이 들어갈 수 있다.
a=[]
b=[1,2,3]
c = ['Life', 'is', 'too', 'short']
d = [1, 2, 'Life', 'is']
e = [1, 2, ['Life', 'is']]
a,b,c,d,e

([],
 [1, 2, 3],
 ['Life', 'is', 'too', 'short'],
 [1, 2, 'Life', 'is'],
 [1, 2, ['Life', 'is']])

In [7]:
##비어 있는 리스트는 a = list()로 생성할 수 있다.
a=list()
a,type(a)

([], list)

### 리스트의 인덱싱

In [8]:
a=[1,2,3]
a

[1, 2, 3]

In [9]:
a[0]

1

In [10]:
a[0]+a[2]

4

In [11]:
a[-1]

3

In [12]:
a = [1, 2, 3, ['a', 'b', 'c']] ##집합안에 집합 총4개의 객체인것이다.

In [13]:
a[0]

1

In [14]:
a[-1]

['a', 'b', 'c']

In [15]:
a[3]

['a', 'b', 'c']

In [19]:
a[-1][0]
## a[-1]이 ['a', 'b', 'c'] 리스트라는 것은 이미 설명했다.
##바로 이 리스트에서 첫 번째 요소를 불러오기 위해 [0]을 붙여 준 것이다.

'a'

In [23]:
##삼중 리스트 인덱싱

a = [1, 2, ['a', 'b', ['Life', 'is']]]
a

[1, 2, ['a', 'b', ['Life', 'is']]]

In [24]:
a[2][2][0]

'Life'

### 리스트의 슬라이싱

In [26]:
a = [1, 2, 3, 4, 5]
a[0:2]

[1, 2]

In [29]:
a = "12345"
a[0:2] ##문자열에서 했던 것과 사용법이 완전히 동일하다.

'12'

In [33]:
a = [1, 2, 3, 4, 5] ##인덱스 슬라이싱 할 때 -1
b = a[:2]
c = a[2:]

In [34]:
b

[1, 2]

In [35]:
c

[3, 4, 5]

### 중첩된 리스트에서 슬라이싱하기

In [38]:
a = [1, 2, 3, ['a', 'b', 'c'], 4, 5]
a[2:5]

[3, ['a', 'b', 'c'], 4]

In [39]:
a[3][:2]

['a', 'b']

### del 함수를 사용해 리스트 요소 삭제하기

In [41]:
a = [1, 2, 3]
del a[1]
a

[1, 3]

In [43]:
a = [1, 2, 3, 4, 5]
del a[2:]
a

[1, 2]

### 리스트에 요소 추가하기 - append << 맨뒤에 추가하는거

In [52]:
a = [1, 2, 3]
a.append(4)
a

[1, 2, 3, 4]

### 리스트 정렬 - sort

In [53]:
##sort 함수는 리스트의 요소를 순서대로 정렬

a = [1, 4, 3, 2]
a.sort()
a

[1, 2, 3, 4]

### 리스트 뒤집기 - reverse

In [54]:
##reverse 함수는 리스트를 역순으로 뒤집어 준다.
a = ['a', 'c', 'b']
a.reverse()
a

['b', 'c', 'a']

### 인덱스 반환 - index

In [51]:
a = [1, 2, 3]
a.index(3)

2

### 리스트에 요소 삽입 - insert

In [55]:
##insert(a, b)는 리스트의 a번째 위치에 b를 삽입하는 함수

a = [1, 2, 3]
a.insert(0, 4)
a

[4, 1, 2, 3]

### 리스트 요소 제거 - remove

In [60]:
##remove(x)는 리스트에서 첫 번째로 나오는 x를 삭제하는 함수
a = [1, 2, 3, 1, 2, 3]
a.remove(3)
a

[1, 2, 1, 2, 3]

In [61]:
a.remove(3)
a

[1, 2, 1, 2]

### 리스트 요소 끄집어 내기 - pop

In [62]:
##pop()은 리스트의 맨 마지막 요소를 리턴하고 그 요소는 삭제한다.
a = [1, 2, 3]
a.pop()

3

### 리스트에 포함된 요소 x의 개수 세기 - count

In [69]:
a = [1, 2, 3, 1]
a.count(1)

2

### 리스트 확장 - extend

In [70]:
##extend(x)에서 x에는 리스트만 올 수 있다.
a = [1, 2, 3]
a.extend([4, 5])
a

[1, 2, 3, 4, 5]

In [71]:
##a.extend([4, 5])는 a += [4, 5]와 동일하다.

a += [4,5]
a

[1, 2, 3, 4, 5, 4, 5]

In [72]:
##a += [4, 5]는 a = a + [4, 5]와 동일한 표현식이다.

a= a+[9,9]
a

[1, 2, 3, 4, 5, 4, 5, 9, 9]

# 튜플 자료형

#### 튜플(tuple)은 몇 가지 점을 제외하곤 리스트와 거의 비슷하며 리스트와 다른 점은 다음과 같다.

리스트는 [], 튜플은 ()으로 둘러싼다.   
리스트는 요솟값의 생성, 삭제, 수정이 가능하지만
#### 튜플은 요솟값을 바꿀 수 없다. (정책적으로 변경을 할 수 없는 요솟값들 )

In [74]:
t1 = ()
t2 = (1,)
t3 = (1, 2, 3)
t4 = 1, 2, 3
t5 = ('a', 'b', ('ab', 'cd'))

In [75]:
t1,t2,t3,t4,t5

((), (1,), (1, 2, 3), (1, 2, 3), ('a', 'b', ('ab', 'cd')))

In [76]:
type(t1)

tuple

In [78]:
type((1)), type((1,))

(int, tuple)

### 튜플 삭제/수정 시도 (당연히 안됌)

In [79]:
t1 = (1, 2, 'a', 'b')
del t1[0]

TypeError: 'tuple' object doesn't support item deletion

In [80]:
t1 = (1, 2, 'a', 'b')
t1[0] = 'c'

TypeError: 'tuple' object does not support item assignment

## 튜플 다루기

### 인덱싱하기

In [81]:
t1 = (1, 2, 'a', 'b')
t1[0]

1

In [82]:
t1[3]

'b'

### 슬라이싱하기

In [83]:
t1 = (1, 2, 'a', 'b')
t1[1:]

(2, 'a', 'b')

### 튜플 더하기

In [84]:
t1 = (1, 2, 'a', 'b')
t2 = (3, 4)
t3 = t1 + t2
t3

(1, 2, 'a', 'b', 3, 4)

### 튜플 곱하기

In [86]:
t2 = (3, 4)
t3 = t2 * 3
t3

(3, 4, 3, 4, 3, 4)

### 튜플 길이 구하기

In [87]:
t1 = (1, 2, 'a', 'b')
len(t1)

4

# 딕셔너리 자료형

사람은 누구든지 "이름" = "홍길동", "생일" = "몇 월 며칠" 등과 같은 방식으로 그 사람이 가진 정보를 나타낼 수 있다. 파이썬은 영리하게도 이러한 대응 관계를 나타낼 수 있는 딕셔너리(dictionary) 자료형을 가지고 있다. 이번에는 이 자료형에 대해 알아보자.

##### 요즘 사용하는 대부분의 언어도 이러한 대응 관계를 나타내는 자료형을 가지고 있는데 이를 딕셔너리라고 하고, ‘연관 배열(associative array)’또는 ‘해시(hash)’라고도 한다.

##### 딕셔너리는 리스트나 튜플처럼 순차적으로(sequential) 해당 요솟값을 구하지 않고 Key를 통해 Value를 얻는다.

In [89]:
dic = {'name': 'pey', 'phone': '010-9999-1234', 'birth': '1118'}
dic

{'name': 'pey', 'phone': '010-9999-1234', 'birth': '1118'}

In [91]:
a = {1: 'hi'}
a

{1: 'hi'}

In [92]:
a = {'a': [1, 2, 3]}
a

{'a': [1, 2, 3]}

### 딕셔너리 쌍 추가, 삭제하기

In [93]:
#딕셔너리 쌍 추가하기
a = {1: 'a'}
a[2] = 'b'
a

{1: 'a', 2: 'b'}

In [97]:
a['name'] = 'pey'
a

{1: 'a', 2: 'b', 3: [1, 2, 3], 'name': 'pey'}

In [98]:
a[3] = [1, 2, 3]
a

{1: 'a', 2: 'b', 3: [1, 2, 3], 'name': 'pey'}

In [99]:
#딕셔너리 요소 삭제하기
del a[1]
a

{2: 'b', 3: [1, 2, 3], 'name': 'pey'}

### 딕셔너리에서 Key를 사용해 Value 얻기

In [100]:
grade = {'pey': 10, 'julliet': 99}
grade['pey']

10

In [101]:
grade['julliet']

99

### 딕셔너리 만들 때 주의할 사항

### 딕셔너리 관련 함수

In [102]:
#key
a1 = 'str'
a2 = [1,2,3]
a3 = (1,)
a4 = {1:'a'}
type(a1),type(a2),type(a3),type(a4)

(str, list, tuple, dict)

In [103]:
a = {'name': 'pey', 'phone': '010-9999-1234', 'birth': '1118'}
a.keys()

dict_keys(['name', 'phone', 'birth'])

In [104]:
for k in a.keys():
    print(k)

name
phone
birth


In [106]:
#Value 리스트 만들기 - values 
a.values()

dict_values(['pey', '010-9999-1234', '1118'])

In [107]:
#Key, Value 쌍 얻기 - items
a.items()

dict_items([('name', 'pey'), ('phone', '010-9999-1234'), ('birth', '1118')])

In [108]:
#get

a = {'name': 'pey', 'phone': '010-9999-1234', 'birth': '1118'}
a.get('name')

'pey'

In [109]:
a.get('phone')

'010-9999-1234'

In [110]:
a = {'name':'pey', 'phone':'010-9999-1234', 'birth': '1118'}
print(a.get('nokey'))

None


In [111]:
print(a['nokey’])

SyntaxError: unterminated string literal (detected at line 1) (440620461.py, line 1)

In [112]:
a.get('nokey', 'foo')

'foo'

In [113]:
#해당 Key가 딕셔너리 안에 있는지 조사하기 - in

a = {'name':'pey', 'phone':'010-9999-1234', 'birth': '1118'}
'name' in a

True

In [114]:
'email' in a

False

# 집합 자료형
### 중복을 허용하지 않는다. 순서가 없다(Unordered).

set은 중복을 허용하지 않는 특징 때문에 데이터의 중복을 제거하기 위한 필터로 종종 사용된다.

In [117]:
s1 = set([1, 2, 3])
s1, type(s1)

({1, 2, 3}, set)

In [119]:
s2 = set("Hello")
s2, type(s2)

({'H', 'e', 'l', 'o'}, set)

### 만약 set 자료형에 저장된 값을 인덱싱으로 접근하려면 다음과 같이 리스트나 튜플로 변환한 후에 해야 한다.

In [120]:
>>> s1 = set([1, 2, 3])
>>> l1 = list(s1)
>>> l1

[1, 2, 3]

In [121]:
l1[0]

1

In [122]:
>>> t1 = tuple(s1)
>>> t1

(1, 2, 3)

In [123]:
>>> t1[0]

1

### 교집합, 합집합, 차집합 구하기

In [125]:
s1 = set([1, 2, 3, 4, 5, 6])
s2 = set([4, 5, 6, 7, 8, 9])

In [128]:
##교집합

s1 & s2 

{4, 5, 6}

In [130]:
s1.intersection(s2) #함수

{4, 5, 6}

In [129]:
#합집합

s1 | s2

{1, 2, 3, 4, 5, 6, 7, 8, 9}

In [131]:
s1.union(s2) #함수

{1, 2, 3, 4, 5, 6, 7, 8, 9}

In [132]:
#차집합

s1 - s2

{1, 2, 3}

In [133]:
s2 - s1

{7, 8, 9}

In [134]:
s1.difference(s2) #함수

{1, 2, 3}

In [135]:
s2.difference(s1)

{7, 8, 9}

### 집합 자료형 관련 함수

In [138]:
#값 1개 추가하기 - add

s1 = set([1, 2, 3])
s1.add(4)
s1

{1, 2, 3, 4}

In [139]:
#값 여러 개 추가하기 - update

s1 = set([1, 2, 3])
s1.update([4, 5, 6])
s1

{1, 2, 3, 4, 5, 6}

In [140]:
#특정 값 제거하기 - remove

s1 = set([1, 2, 3])
s1.remove(2)
s1

{1, 3}

# 불 자료형(논리)

불(bool) 자료형이란 참(True)과 거짓(False)을 나타내는 자료형이다.
- True: 참을 의미한다.
- False: 거짓을 의미한다.   
True나 False는 파이썬의 예약어로, true, false와 같이 작성하면 안 되고 첫 문자를 항상 대문자로 작성해야 한다.

In [141]:
>>> a = True
>>> b = False

In [142]:
type(a), type(b)

(bool, bool)

In [143]:
1 == 1

True

In [144]:
2 > 1

True

In [145]:
2 < 1

False

### 자료형의 참과 거짓

In [ ]:
a = [1, 2, 3, 4]
while a:
    print(a.pop(안녕하세요))

In [148]:
if [1,2,3]:
...     print("참")
else:
...     print("거짓")

참


### 불연산

In [149]:
bool('python')

True

In [150]:
bool('')

False

In [151]:
bool(0)

False

In [152]:
bool(3)

True

In [153]:
bool([1, 2, 3])

True

In [154]:
bool([])

False